pkg/trace/api/info_test.go

```diff
@@ -109,145 +293,56 @@ func TestInfoHandler(t *testing.T) {
-		},
-		"obfuscation": {
-			"elastic_search": true,
-			"mongo": true,
-			"sql_exec_plan": true,
-			"sql_exec_plan_normalize": true,
-			"http": {
-				"remove_query_string": true,
-				"remove_path_digits": true
-			},
-			"remove_stack_traces": false,
-			"redis": true,
-			"memcached": false
-		}
-	}
-}`,
-		},
-		{
-			name: "debug",
-			expected: `{
-	"version": "0.99.0",
-	"git_commit": "fab047e10",
-	"endpoints": [
-		"/v0.3/traces",
-		"/v0.3/services",
-		"/v0.4/traces",
-		"/v0.4/services",
-		"/v0.5/traces",
-		"/v0.7/traces",
-		"/profiling/v1/input",
-		"/telemetry/proxy/",
-		"/v0.6/stats",
-		"/v0.1/pipeline_stats",
-		"/appsec/proxy/",
-		"/evp_proxy/v1/",
-		"/debugger/v1/input"
-		"/dogstatsd/v1/proxy"
-	],
-	"feature_flags": [
-		"feature_flag"
-	],
-	"client_drop_p0s": true,
-	"span_meta_structs": true,
-	"long_running_spans": true,
-	"config": {
-		"default_env": "prod",
-		"target_tps": 11,
-		"max_eps": 12,
-		"receiver_port": 8111,
-		"receiver_socket": "/sock/path",
-		"connection_limit": 12,
-		"receiver_timeout": 100,
-		"max_request_bytes": 123,
-		"statsd_port": 123,
-		"max_memory": 1000000,
-		"max_cpu": 12345,
-		"analyzed_spans_by_service": {
-			"X": {
-				"Y": 2.4
-			}
+	expectedKeys := map[string]interface{}{
+		"version":            nil,
+		"git_commit":         nil,
+		"endpoints":          nil,
+		"feature_flags":      nil,
+		"client_drop_p0s":    nil,
+		"span_meta_structs":  nil,
+		"long_running_spans": nil,
+		"config": map[string]interface{}{
+			"default_env":               nil,
+			"target_tps":                nil,
+			"max_eps":                   nil,
+			"receiver_port":             nil,
+			"receiver_socket":           nil,
+			"connection_limit":          nil,
+			"receiver_timeout":          nil,
+			"max_request_bytes":         nil,
+			"statsd_port":               nil,
+			"max_memory":                nil,
+			"max_cpu":                   nil,
+			"analyzed_spans_by_service": nil,
+			"obfuscation": map[string]interface{}{
+				"elastic_search":          nil,
+				"mongo":                   nil,
+				"sql_exec_plan":           nil,
+				"sql_exec_plan_normalize": nil,
+				"http": map[string]interface{}{
+					"remove_query_string": nil,
+					"remove_path_digits":  nil,
+				},
+				"remove_stack_traces": nil,
+				"redis":               nil,
+				"memcached":           nil,
+			},
 		},
-		"obfuscation": {
-			"elastic_search": true,
-			"mongo": true,
-			"sql_exec_plan": true,
-			"sql_exec_plan_normalize": true,
-			"http": {
-				"remove_query_string": true,
-				"remove_path_digits": true
-			},
-			"remove_stack_traces": false,
-			"redis": true,
-			"memcached": false
-		}
 	}
-}`,
-		},
+
+	rcv := newTestReceiverFromConfig(conf)
+	defer testutil.WithFeatures("feature_flag")()
+	_, h := rcv.makeInfoHandler()
+	rec := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "/info", nil)
+	h.ServeHTTP(rec, req)
+	var m map[string]interface{}
+	b, err := ioutil.ReadAll(rec.Body)
```

Comment:
Super nit:

You can save a few lines by doing `json.NewDecoder(rec.Body).Decode(&m)`

Reply:
Nice, you're right.